blobfile/_auto_parallelizer.py

```diff
@@ -0,0 +1,118 @@
+"""
+Module for automatically parallelizing a function that takes a long time to run but can be split into smaller tasks.
+
+See doc string on `parallelize` for more details.
+"""
+from typing import Callable, Any, Generator, List
+import concurrent.futures
+import time
+from dataclasses import dataclass
+import multiprocessing
+from threading import Event
+
+@dataclass
+class _FuncResult:
+    did_complete: bool
+    result: Any
+
+@dataclass
+class _CancellableFuncCall:
```

Comment:
Maybe there's a better way to do this? I know there's a failure case here if the task doesn't give up the GIL.